src/sentry/snuba/metrics/query_builder.py

```diff
@@ -587,9 +589,21 @@ def _extract_data(self, data, groups):
                         if series[series_index] == default_null_value:
                             series[series_index] = cleaned_value
 
-    def translate_results(self):
+    def _get_operation_with_public_name(self, operation: Optional[str], metric_mri: str) -> str:
+        if operation is None:
+            return get_public_name_from_mri(metric_mri)
+        return f"{operation}({get_public_name_from_mri(metric_mri)})"
+
+    def _parse_operation_and_metric_mri(self, name: str) -> Tuple[Optional[str], str]:
+        matches = MRI_EXPRESSION_REGEX.match(name)
+        if matches:
+            # operation, metric_mri
+            return matches[1], matches[2]
```

Comment:
`if MRI_NAME_REGEX.matches....`

Reply:
after deep dive: we can't use `MRI_NAME_REGEX = r"([a-z_]+(?:\\.[a-z_]+)*)"` here it breaks things https://github.com/getsentry/sentry/blob/9441adfc40a7923bbbf1c663988ad15c26511abb/src/sentry/snuba/metrics/query_builder.py#L683

This function should only accept metric with operation or fall back to metric without operation.